calculator.js

```diff
@@ -2,4 +2,174 @@ const exampleAdditionInput = {
   num1: 3,
   num2: 5,
   operation: 'add',
-}
+};
+
+const exampleAdditionInput2 = {
+  num1: 3,
+  num2: 5,
+  operation: '+',
+};
+
+const exampleSubractionInput = {
+  num1: 9,
+  num2: 3,
+  operation: 'subtract',
+};
+
+const exampleSubractionInput2 = {
+  num1: 9,
+  num2: 3,
+  operation: '-',
+};
+
+const exampleMultiplicationInput = {
+  num1: 12,
+  num2: 23,
+  operation: 'multiply',
+};
+
+const exampleMultiplicationInput2 = {
+  num1: 12,
+  num2: 23,
+  operation: '*',
+};
+
+const exampleDivisionInput = {
+  num1: 13,
+  num2: 4,
+  operation: 'divide',
+};
+
+const exampleDivisionInput2 = {
+  num1: 13,
+  num2: 4,
+  operation: '/',
+};
+
+const exampleDiv0ErrorInput = {
+  num1: 13,
+  num2: 0,
+  operation: '/',
+};
+
+const exampleInvalidOperationInput = {
+  num1: 13,
+  num2: 4,
+  operation: 'quarantine',
+};
+
+const exampleInvalidNumInput = {
+  num1: 'horse',
+  num2: 23,
+  operation: '-',
+};
+
+const exampleInvalidNumInput2 = {
+  num1: 13,
+  num2: 'horse',
+  operation: '*',
+};
+
+const exampleInvalidNumInput3 = {
+  num1: 'horse',
+  num2: 'buggy',
+  operation: 'divide',
+};
+
+const calculate = function(input) { 
+  if(isNaN(input.num1) || isNaN(input.num2) || input.num1 === "" || input.num1 === " " || input.num2 === "" || input.num2 === " "){
+    if(isNaN(input.num1) && isNaN(input.num2)){
+      console.log(`${input.num1} and ${input.num2} are not numbers. Please enter numbers.`);
+    }
+    else if(isNaN(input.num1)){
+      console.log(`${input.num1} is not a numbers. Please enter a number.`);
+    }
+    else if(isNaN(input.num2)){
+      console.log(`${input.num2} is not a number. Please enter a number.`);
+    };
+  }
+  else{
+
+    if(input.operation === 'add' || input.operation === '+'){
+      let sum = input.num1 + input.num2;
+      return sum;
+    }
+    else if(input.operation === 'subtract' || input.operation === '-'){
+      let difference = input.num1 - input.num2;
+      return difference;
+    }
+    else if(input.operation === 'multiply' || input.operation === '*'){
+      let product = input.num1 * input.num2;
```

Comment:
Could the variables `sum`, `difference` and `product` be declared with `const` here instead of `let`? It doesn't look like you expect their values to change.